EventProc/src/HcalDigiProducer.cxx

```diff
@@ -78,14 +88,22 @@ namespace ldmx {
             }           
 
             if (verbose_) {
-                std::cout << "detIDraw: " << detIDraw << std::endl;
+                std::cout << "section: " << detID_->getFieldValue("section") << "  layer: " << detID_->getFieldValue("layer") <<  "  strip: " << detID_->getFieldValue("strip") <<std::endl;
+            }        
+
+            // re-assign the strip number based on super strip size -- ONLY FOR Back Hcal
+            // int subsection = detID_->getFieldValue("section");
+            if (SUPER_STRIP_SIZE_ != 1 && subsection == 0){
+                int detIDraw_orig = detIDraw;
                 detID_->setRawValue(detIDraw);
                 detID_->unpack();
-                int layer = detID_->getFieldValue("layer");
-                int subsection = detID_->getFieldValue("section");
                 int strip = detID_->getFieldValue("strip");                
-                std::cout << "section: " << subsection << "  layer: " << layer <<  "  strip: " << strip <<std::endl;
-            }           
+                int newstrip = strip/SUPER_STRIP_SIZE_;
+                detID_->setFieldValue("strip",newstrip);
+                detID_->pack();
+                // get the new raw value
+                detIDraw = detID_->getRawValue();
+            }
             
             // for now, we take am energy weighted average of the hit in each stip to simulate the hit position. 
             // will use strip TOF and light yield between strips to estimate position.            
```

Comment:
I think this is the right place to add the hit position filtering if we want it, since we still have the correct location of the deposited energy. The HCAL veto already has a averaged / smeared position, so this would be less reliable to put it in there.

Reply:
Do you mean if we allow a hit or not if we want to restrict the size of the hcal for different vetoes?  If yes, it's an interesting issue of where to do this.  I would probably make a check a little later down in the code around L111 then you would have multiple digiProducers for multiple vetoes.  Is that what you were thinking?